builtin/logical/aws/secret_access_keys_test.go

```diff
@@ -40,3 +41,51 @@ func TestNormalizeDisplayName_NormNotRequired(t *testing.T) {
 		}
 	}
 }
+
+func TestGenUsername(t *testing.T) {
+
+	testUsername, warning := genUsername("name1", "policy1", "iam_user", `{{ printf "vault-%s-%s-%s-%s" (.DisplayName) (.PolicyName) (unix_time) (random 20) | truncate 64 }}`)
+	if warning != "" {
+		t.Fatalf(
+			"expected no warning; got %s",
+			warning,
+		)
+	}
+	if !strings.HasPrefix(testUsername, "vault-name1-policy1") {
+		t.Fatalf(
+			"expected return to match template, got %s",
+			testUsername,
+		)
+	}
+	// IAM usernames are capped at 64 characters
+	if len(testUsername) > 64 {
+		t.Fatalf(
+			"expected IAM username to be of length 64, got %d",
+			len(testUsername),
+		)
+	}
```

Comment:
Can this be a regex assertion? This is the assertion library I used for MongoDB Atlas: https://pkg.go.dev/github.com/stretchr/testify/require?utm_source=godoc#Regexp

And here is an example usage: https://github.com/hashicorp/vault-plugin-database-mongodbatlas/pull/26/files#diff-e1827d78ff762e88c62c9f7c03dcadd7e5db3104d97d6ff4e1336eb8ef4635d1R241-R242

Reply:
Fixed!